doc/src/manual/types.md

```diff
@@ -567,26 +562,34 @@ See [this FAQ entry](@ref faq-nothing) for more information.
 
 An important and powerful feature of Julia's type system is that it is parametric: types can take
 parameters, so that type declarations actually introduce a whole family of new types -- one for
-each possible combination of parameter values. There are many languages that support some version
-of [generic programming](https://en.wikipedia.org/wiki/Generic_programming), wherein data structures
-and algorithms to manipulate them may be specified without specifying the exact types involved.
-For example, some form of generic programming exists in ML, Haskell, Ada, Eiffel, C++, Java, C#,
-F#, and Scala, just to name a few. Some of these languages support true parametric polymorphism
-(e.g. ML, Haskell, Scala), while others support ad-hoc, template-based styles of generic programming
-(e.g. C++, Java). With so many different varieties of generic programming and parametric types
-in various languages, we won't even attempt to compare Julia's parametric types to other languages,
-but will instead focus on explaining Julia's system in its own right. We will note, however, that
-because Julia is a dynamically typed language and doesn't need to make all type decisions at compile
-time, many traditional difficulties encountered in static parametric type systems can be relatively
-easily handled.
-
-All declared types (the `DataType` variety) can be parameterized, with the same syntax in each
-case. We will discuss them in the following order: first, parametric composite types, then parametric
-abstract types, and finally parametric primitive types.
+each possible combination of parameter values. This is often referred to as "generic programming".
+There are two kinds of explicitly declared parametric types.
+[Parametric composite types](#man-parametric-composite-types)
+declare a family of concrete, [composite types](#Composite-Types),
+whereas [parametric abstract types](#Parametric-Abstract-Types)
+declare a family of [abstract types](#man-abstract-types).
+Other kinds of built-in parametric types such as tuple types are discussed in this section as well.
+Parametric types can also be used in [type annotations](#Type-Annotations).
+
+!!! note
+    Parametric types do not represent a single node in the type graph.
+    Therefore, [`typeof`](@ref) can never return a parametric type name.
+    To dynamically verify a (parametric) subtype relation, use [`isa`](@ref) instead.
+
+!!! warning
+    Type parameters are invariant: even though `Float64 <: Real`,
+    the relation `Foo{Float64} <: Foo{Real}` does NOT hold for any parametric type `Foo`.
```

Comment:
```suggestion
    the relation `Foo{Float64} <: Foo{Real}` does NOT hold for any parametric type `Foo`.
```

of course except for `Tuple`

Reply:
I didn't know about that, thanks. Is there a reason why that is the case that we can/should highlight here?

Reply:
I'm not familiar with all the details but my understanding is that the reason for covariance is largely vestigial; it made dispatch easier in the early days of Julia or something


Tuple type covariance is described here in the docs https://docs.julialang.org/en/v1/manual/types/#Tuple-Types-1

and there is an open issue discussing the potential to switch to invariance in a hypothetical breaking release of Julia (although as a gambling man, I wouldn't put my money on that change ever happening) https://github.com/JuliaLang/julia/issues/24614

I would think a quick link to the docs is probably sufficient